src/main/java/com/google/api/gax/grpc/ServiceApiSettings.java

```diff
@@ -66,34 +98,51 @@
   /**
    * Constructs an instance of ServiceApiSettings.
    */
-  protected ServiceApiSettings(ManagedChannel channel,
-                               boolean shouldAutoCloseChannel,
-                               ScheduledExecutorService executor,
-                               ConnectionSettings connectionSettings,
-                               String generatorName,
-                               String generatorVersion,
-                               String clientLibName,
-                               String clientLibVersion) {
-    this.channel = channel;
-    this.executor = executor;
-    this.connectionSettings = connectionSettings;
-    this.shouldAutoCloseChannel = shouldAutoCloseChannel;
+  protected ServiceApiSettings(
+      ChannelProvider channelProvider,
+      ExecutorProvider executorProvider,
+      String generatorName,
+      String generatorVersion,
+      String clientLibName,
+      String clientLibVersion) {
+    this.channelProvider = channelProvider;
+    this.executorProvider = executorProvider;
     this.clientLibName = clientLibName;
     this.clientLibVersion = clientLibVersion;
     this.generatorName = generatorName;
     this.generatorVersion = generatorVersion;
   }
 
-  public final ManagedChannel getChannel() {
-    return channel;
+  /**
+   * Return the channel to be used to connect to the service, retrieved using the channelProvider.
+   * If no channel was set, a default channel will be instantiated.
+   */
+  public final ManagedChannel getOrBuildChannel() throws IOException {
+    return getChannelProvider().getChannel(getOrBuildExecutor());
```

Comment:
I think the way this is set up, once a channel has been constructed, the same one will keep being passed out. I was looking to have it not do that, so that:

1. A settings class can be re-used
2. A settings class wouldn't keep providing a channel that had been "transferred" to an XApi class